spec/models/broadcast_spec.rb

```diff
@@ -18,4 +18,14 @@
     expect(inactive_broadcast).not_to be_valid
     expect(inactive_broadcast.errors.full_messages.join).to include("You can only have one active announcement broadcast")
   end
+
+  it "updates the Broadcasts last_active_at timestamp" do
+    Timecop.freeze(Time.current) do
+      current_time = Time.zone.now
+      broadcast = create(:welcome_broadcast, active: false)
+      expect(broadcast.last_active_at).to eq(2.days.ago)
+      broadcast.update(active: true)
+      expect(broadcast.last_active_at).to eq current_time
+    end
+  end
```

Comment:
I think we could also use some tests in `spec/requests/internal/broadcasts_spec.rb` that test that this attribute does/does not update correctly from the controller side

Reply:
I had a spec for this in `/internal/broadcasts_spec.rb` initially, but after speaking with @atsmith813 on Friday, he suggested that I move my spec to the Broadcast model since the method that updates the timestamp is on the model itself. That being said, is it necessary and does it make sense to add a spec to `/internal/broadcasts_spec.rb` in addition to the one in the model? cc @atsmith813

Reply:
The test that you have _does_ test that the model behavior works; but what happens if you remove `:active` from [here](https://github.com/thepracticaldev/dev.to/pull/8451/files#diff-08d4f0af1a44965d340da617d97718dbR59)?

In fact, I'd try it out; if you removed it from the params, your [test](https://github.com/thepracticaldev/dev.to/pull/8451/files#diff-37ba91f757769b082f98c5b68d981c8fR22) on line 22 would still pass, since that test is manually updating the attribute. You're testing that the model behavior works, but not testing that the attribute can be updated from the controller/via a request.

That's why I'd advocate for adding a test to make sure that you can make a request to change this attribute from the controller side. That way, if something about the controller breaks, you'll catch the regression (even if the model side changes aren't affected).

Does that make sense?

Reply:
Yes, that makes sense. I have added a spec for this per your request. 😄